src/azul/indexer/index_service.py

```diff
@@ -717,15 +720,19 @@ def _select_latest(self,
                         cur_bundle_uuid, cur_bundle_version, cur_entity = \
                             collated_entities.get(entity_id, (None, '', None))
                         if cur_entity is not None and entity.keys() != cur_entity.keys():
-                            symmetric_difference = set(entity.keys()).symmetric_difference(cur_entity)
-                            log.warning('Document shape of `%s` entity `%s` does not match between bundles '
-                                        '%s, version %s and %s, version %s: %s',
-                                        entity_type, entity_id,
-                                        cur_bundle_uuid, cur_bundle_version,
-                                        contribution.coordinates.bundle.uuid,
-                                        contribution.coordinates.bundle.version,
-                                        symmetric_difference)
-                        if cur_bundle_version < contribution.coordinates.bundle.version:
+                            if cur_bundle_version == contribution.coordinates.bundle.version:
+                                assert contribution.entity.entity_type == 'datasets', contribution
+                                entity = deep_dict_merge((entity, cur_entity))
+                            else:
+                                symmetric_difference = set(entity.keys()).symmetric_difference(cur_entity)
+                                log.warning('Document shape of `%s` entity `%s` does not match between bundles '
+                                            '%s, version %s and %s, version %s: %s',
+                                            entity_type, entity_id,
+                                            cur_bundle_uuid, cur_bundle_version,
+                                            contribution.coordinates.bundle.uuid,
+                                            contribution.coordinates.bundle.version,
+                                            symmetric_difference)
```

Comment:
This should be a separate, first commit. I want to be extra sure it doesn't break HCA.